skimage/transform/_thin_plate_splines.py

```diff
@@ -0,0 +1,147 @@
+import numpy as np
+from scipy import ndimage
+
+
+def warp_images(
+    from_points,
+    to_points,
+    images,
+    output_region,
+    interpolation_order=1,
+    approximate_grid=2
+):
+    """Return an array of warped images.
+
+    Define a thin-plate-spline warping transform that warps from the
+    from_points to the to_points, and then warp the given images by
+    that transform.
+
+    Parameters
+    ----------
+    from_points: (N, 2) array_like
+        Source image coordinates.
+    to_points: (N, 2) array_like
+        Target image coordinates.
+    images: ndarray
+        Array of nD images to be warped with the given warp transform.
+    output_region: (1, 4) array
+        The (xmin, ymin, xmax, ymax) region of the output
+        image that should be produced. (Note: The region is inclusive, i.e.
+        xmin <= x <= xmax)
+    interpolation_order: int, optional
+        If value is 1, then use linear interpolation else use
+        nearest-neighbor interpolation.
+    approximate_grid: int, optional
+        If approximate_grid is greater than 1, say x, then the transform is
+        defined on a grid x times smaller than the output image region.
+        Then the transform is bilinearly interpolated to the larger region.
+        This is fairly accurate for values up to 10 or so.
+
+    Returns
+    -------
+    warped: array_like
+        Array of warped images.
+
+    References
+    ----------
+    .. [1] Bookstein, Fred L. "Principal warps: Thin-plate splines and the
+    decomposition of deformations." IEEE Transactions on pattern analysis and
+    machine intelligence 11.6 (1989): 567–585.
+
+    """
+    transform = _make_inverse_warp(from_points, to_points,
+                                   output_region, approximate_grid)
+    return [ndimage.map_coordinates(np.asarray(image), transform, order=interpolation_order) for image in images]
+
+
+def _make_inverse_warp(
+        from_points, to_points, output_region,
+        approximate_grid):
+    x_min, y_min, x_max, y_max = output_region
+    if approximate_grid is None:
+        approximate_grid = 1
+    x_steps = (x_max - x_min) // approximate_grid
+    y_steps = (y_max - y_min) // approximate_grid
+    x, y = np.mgrid[x_min:x_max:x_steps*1j, y_min:y_max:y_steps*1j]
+
+    # make the reverse transform warping from the to_points to the from_points,
+    # because we do image interpolation in this reverse fashion
+    transform = _make_warp(to_points, from_points, x, y)
+
+    if approximate_grid != 1:
+        # linearly interpolate the zoomed transform grid
+        new_x, new_y = np.mgrid[x_min:x_max+1, y_min:y_max+1]
+        x_fracs, x_indices = np.modf(
+            (x_steps-1)*(new_x-x_min)/float(x_max-x_min))
```

Comment:
If I understand https://github.com/scikit-image/scikit-image/issues/2429#issuecomment-816402127 correctly, we should be able to replace this part with `map_coordinates`? Does someone want to give it a try?

Reply:
Oh, or even [`skimage.transform.rescale`](https://scikit-image.org/docs/dev/api/skimage.transform.html#skimage.transform.rescale). :D

Reply:
That is `scipy.ndimage. map_coordinates `?

Reply:
Yes. We should be able to use it to resample the reverse transform. :)